### DISCUSSION TITLE
Clear breadcrumbs in BeforeSend

### DISCUSSION THREAD

**Discussion by hheexx** [getsentry/sentry-dotnet#2100] [Q&A] [answered]
Hi,

I'm trying to remove breadcrumbs for some events as they are useless and event is very high volume.
Probably best way to do that is BeforeSend but i find most fields there on SentryEvent to be readonly.

I can just AddBreacrumb but I can't find way to remove them.

**Comment by mattjohnsonpint** [comment, marked as answer]
Hi.  Thanks for the feedback.  Indeed, we need to add an API to clear breadcrumbs.  See https://github.com/getsentry/sentry-dotnet/issues/1660#issuecomment-1347269224

**Reply by hheexx** [reply]
Thanks for info.

Big parts of SentryEvent are readonly. I'm not sure is there a reason to be like that.
But it really cripples BeforeSend function.
